nautobot/core/api/serializers.py

```diff
@@ -564,9 +595,14 @@ def _save_relationships(self, instance, relationships):
                     remove_assoc.delete()
 
     def get_field_names(self, declared_fields, info):
-        """Ensure that "relationships" is always included as an opt-in field."""
+        """Ensure that "relationships" is included as an opt-in field on root serializers."""
         fields = list(super().get_field_names(declared_fields, info))
-        self.extend_field_names(fields, "relationships", opt_in_only=True)
+        if not self.is_nested:
+            # Only include relationships as opt-in.
+            self.extend_field_names(fields, "relationships", opt_in_only=True)
+        else:
+            # As relationships are expensive, do not include them on nested serializers even if opted in.
+            fields.remove("relationships")
```

Comment:
Same as above, should this account for `ValueError` if `relationships` does not exist in `fields`?